php-plugin/src/main/java/fr/greencodeinitiative/php/checks/AvoidGettingSizeCollectionInForLoopCheck.java

```diff
@@ -0,0 +1,65 @@
+package fr.greencodeinitiative.php.checks;
+
+import org.sonar.check.Priority;
+import org.sonar.check.Rule;
+import org.sonar.plugins.php.api.tree.SeparatedList;
+import org.sonar.plugins.php.api.tree.Tree;
+import org.sonar.plugins.php.api.tree.expression.BinaryExpressionTree;
+import org.sonar.plugins.php.api.tree.expression.ExpressionTree;
+import org.sonar.plugins.php.api.tree.expression.ParenthesisedExpressionTree;
+import org.sonar.plugins.php.api.tree.statement.*;
+import org.sonar.plugins.php.api.visitors.PHPSubscriptionCheck;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.regex.Pattern;
+
+@Rule(
+        key = AvoidGettingSizeCollectionInForLoopCheck.RULE_KEY,
+        name = AvoidGettingSizeCollectionInForLoopCheck.ERROR_MESSAGE,
+        description = AvoidGettingSizeCollectionInForLoopCheck.ERROR_MESSAGE,
+        priority = Priority.MINOR,
+        tags = {"bug", "eco-design", "ecocode"})
```

Comment:
delete `bug` because not a "bug" (some existing rules with `bug` will be checked again)